python/cudf/cudf/core/column/string.py

```diff
@@ -2038,7 +2038,12 @@ def as_numerical_column(self, dtype, **kwargs):
         kwargs.update(dtype=out_dtype)
 
         if str_dtype.kind in ("i"):
-            if not cpp_all_integers(self):
+            if self.null_count > 0:
+                string_col = self.dropna()
+            else:
+                string_col = self
+
+            if not cpp_all_integers(string_col):
```

Comment:
It would be great if we moved this down to a parameter for the libcudf API. I imagine we could return True for nulls instead of False relatively easily and it would save us having to effectively do a stream compaction here. cc @davidwendt 

Not a blocker for merging though for sure.

Reply:
Adding parameter to `cudf::strings::all_integers()` to have it return true for nulls would not be difficult.

Reply:
Sure, in that case, should we hold off this PR until those changes are in, or should we adapt to the changes in a follow-up PR?

@davidwendt, we'll also need the same parameter for `all_floats` API too as we are handling the null case in a similar fashion too.

Reply:
No need to hold off, @galipremsagar can you write up an issue for adding a parameter to `all_integers()` / `all_floats()`?

Reply:
Yep, link to the issue: https://github.com/rapidsai/cudf/issues/5130